### PR TITLE
{Doc} Refine extension release notes

### PR DESCRIPTION
**Pull request by haroldrandom**
Peer of https://github.com/Azure/azure-cli/pull/13960

---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
add to S171

File: README.md [hunk 1]
```diff
@@ -33,6 +33,18 @@ Add your extension to the index to make it available in these CLI commands:
 - Your extension artifact (i.e. `.whl`) will not live in this repository. You can publish your extension to PyPI or somewhere else such as Azure Storage.
 - If you want your extension to appear in the index.json, modify the index.
 
+## About extension publishing
+
+There is a pipeline to automatically help build, upload and publish extension wheels.
```
Comment (jiasli) [inline README.md:38]:
```suggestion
There is a pipeline to automatically build, upload and publish extension wheels.
```

File: README.md [hunk 2]
```diff
@@ -33,6 +33,18 @@ Add your extension to the index to make it available in these CLI commands:
...
+
+Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.
+
+The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.
```
Comment (jiasli) [inline README.md:42]:
```suggestion
The precondition is to put your code inside this repo and upgrade the version in the PR but not to modify `src/index.json`.
```

File: README.md [hunk 3]
```diff
@@ -33,6 +33,18 @@ Add your extension to the index to make it available in these CLI commands:
...
+
+If you want to host source code in your dedicated repo, you have to upload and WHL file and update the `src/index.json` manually.
```
Comment (jiasli) [inline README.md:44]:
```suggestion
If you want to host the source code in your dedicated repo, you have to upload the WHL file and update the `src/index.json` manually.
```

**Pull request merged**